### ISSUE TITLE
Replacement of relative path in `[sources]` section

### ISSUE THREAD

**Issue by hexaeder** [JuliaLang/Pkg.jl#4023]
Hello,

I really like the new `[sources]` blocks. However, I noticed that some PKG operations transform relative paths into absolute paths.

For example, for my doc environment I'd like to put
```
[sources]
MyPkg = {path = ".."}
```
into `MyPkg/docs/Project.toml`.
However, after some local doc build/tests, this sometimes gets transformed into
```
[sources]
MyPkg = {path = "/home/hw/.juila/dev/MyPkg"}
```
which is far less portable. I think its only a matter of time until people will commit and push that...

**Comment by KristofferC** [comment]
I can't really reproduce this. Do you have a concrete way how to provoke this behavior.

**Comment by goerz** [comment]
Doesn’t https://github.com/JuliaLang/Pkg.jl/issues/4426#issue-3528858238 reproduce this?

And, as I say there, to me the important consideration is to be maximally conservative when modifying an existing `Project.toml`, so things don’t accidentally get committed to git

**Comment by KristofferC** [comment]
> Doesn’t https://github.com/JuliaLang/Pkg.jl/issues/4426#issue-3528858238 reproduce this?

I mean, in there you explicitly put an absolute path there yourself (by your own free will 😆). This talks about

> However, after some local doc build/tests, this sometimes gets transformed into

which suggests it is happening automatically somehow which would be a bug.

**Comment by goerz** [comment]
Fair enough :-)

(and yes, I will also look into modifying my script to no longer call `Pkg.develop` the same way)

**Comment by hexaeder** [comment]
I tried to reproduce it. I think the problem was exactly as written in #4426. The reason I was confused by that is the following:

Lets say i have
```
Project.toml
test/Project.toml
```
In the test project, I have
```
[sources]
MainPackage = {path = ".."}
```

For compatibility with versions prior to 1.11, I had explicit `Pkg.develop(PackageSpec(path=dirname(@__DIR__)))` in some scripts.
This will update the sources block if there is one, but it won't add a sources block if there is none. This feels inconsistent to me. Also, whats the difference between add/dev in that context? I could understand if `Pkg.add` + path/url **always** adds/modifies the [sources] block, but whether a package is in dev mode feels very env-local and thus like a pure Manifest property in my (admittedly naive) logic on how those things work.

Of course, feel free to close this if you don't consider the current behavior inconsistent, I will just update my scripts to not use `Pkg.develop` on 1.11+.

**Comment by KristofferC** [comment]
Yeah, I'll close this then. If you do an explicit develop with an absolute path I think we need to update the path to that.
